### ISSUE TITLE
Dimmable light card doesn't change state

### ISSUE THREAD

**Issue by akalikin1** [anotherjulien/MyHOME#22]
Dimmable light card doesn't change state when toggle on
When changing brightness it works
When toggle off it works

Light and dimmer itself physically works as expected

![dimmable light](https://user-images.githubusercontent.com/51901027/134490264-30a3aea3-35b8-4962-a8a3-de68f191aa46.gif)
.

**Comment by anotherjulien** [comment]
Huh, that's a new one!
Can you tell me what dimmer actuator model you use?
Do you know if there's anything special to it?

**Comment by akalikin1** [comment]
I have 418 dimmer relay, and nothing special, I use node red also and Myhome node, it works like a charm

**Comment by anotherjulien** [comment]
OK, I have both an F418 and F418U2 working without issue at home.
Can you collect debug logs to see what messages are sent and received when you switch ON?

**Comment by akalikin1** [comment]
> OK, I have both an F418 and F418U2 working without issue at home.
> Can you collect debug logs to see what messages are sent and received when you switch ON?

```
2021-09-23 17:53:07 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=light, service=turn_on, service_data=entity_id=light.table_dots>
2021-09-23 17:53:07 DEBUG (MainThread) [custom_components.myhome] Message *1*1*38## was successfully queued.
2021-09-23 17:53:07 DEBUG (MainThread) [custom_components.myhome] Message *#1*38*4## was successfully queued.
2021-09-23 17:53:07 DEBUG (MainThread) [custom_components.myhome] Message *1*1*38## was successfully unqueued by worker 0.
2021-09-23 17:53:07 DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection] [140345772026992] Sending {"id": 44, "type": "result", "success": true, "result": {"context": {"id": "eb233b3d6e44ac11992108cd82c47741", "parent_id": null, "user_id": "627351d1043e48c3bb8a0d679d43614b"}}}
2021-09-23 17:53:07 INFO (MainThread) [custom_components.myhome] Message *1*1*38## was successfully sent.
2021-09-23 17:53:07 DEBUG (MainThread) [custom_components.myhome] Message *#1*38*4## was successfully unqueued by worker 0.
2021-09-23 17:53:07 DEBUG (MainThread) [custom_components.myhome] Message *#1*38*4## was successfully sent.
2021-09-23 17:53:07 DEBUG (MainThread) [custom_components.myhome] Received: *1*9*38##
2021-09-23 17:53:07 DEBUG (MainThread) [custom_components.myhome] Message *#1*38*4## was successfully queued.
2021-09-23 17:53:07 DEBUG (MainThread) [custom_components.myhome] Message *#1*38*4## was successfully unqueued by worker 0.
2021-09-23 17:53:07 DEBUG (MainThread) [custom_components.myhome] Message *#1*38*4## was successfully sent.
2021-09-23 17:53:08 DEBUG (Recorder) [homeassistant.components.recorder.pool.RecorderPool] Connection <sqlite3.Connection object at 0x7fa4d994c8a0> checked out from pool
2021-09-23 17:53:08 DEBUG (Recorder) [homeassistant.components.recorder.pool.RecorderPool] Connection <sqlite3.Connection object at 0x7fa4d994c8a0> being returned to pool
2021-09-23 17:53:08 DEBUG (Recorder) [homeassistant.components.recorder.pool.RecorderPool] Connection <sqlite3.Connection object at 0x7fa4d994c8a0> rollback-on-return
2021-09-23 17:53:10 INFO (MainThread) [homeassistant.components.websocket_api.http.connection] [140345955698576] Connection closed by client
2021-09-23 17:53:10 DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection] [140345955698576] Disconnected
```

**Comment by anotherjulien** [comment]
Ah, I see! So when you click to switch it on, the light actually turns on, right? It's just the switch on the interface that turns back off?
Would you be able to provide the same kind of logs when you set a specific brightness?

**Comment by akalikin1** [comment]
> Ah, I see! So when you click to switch it on, the light actually turns on, right? It's just the switch on the interface that turns back off?

Yes, you right, light turns on with the same brightness, that it was off before, and on the interface it turns back off, as you can see in my first message on attached gif

> Would you be able to provide the same kind of logs when you set a specific brightness?

```
2021-09-23 17:52:58 DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection] [140345772026992] Received {'type': 'call_service', 'domain': 'light', 'service': 'turn_on', 'service_data': {'entity_id': 'light.table_dots', 'brightness_pct': 69}, 'id': 42}
2021-09-23 17:52:58 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=light, service=turn_on, service_data=entity_id=light.table_dots, brightness_pct=69>
2021-09-23 17:52:58 DEBUG (MainThread) [custom_components.myhome] Message *#1*38*#1*169*0## was successfully queued.
2021-09-23 17:52:58 DEBUG (MainThread) [custom_components.myhome] Message *#1*38*#1*169*0## was successfully unqueued by worker 0.
2021-09-23 17:52:58 DEBUG (MainThread) [custom_components.myhome] Command session connection reset, retrying...
2021-09-23 17:52:58 DEBUG (MainThread) [custom_components.myhome] Opening command session.
2021-09-23 17:52:58 DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection] [140345772026992] Sending {"id": 42, "type": "result", "success": true, "result": {"context": {"id": "f6b34d0151a955fe04193d3a7b1cd2c3", "parent_id": null, "user_id": "627351d1043e48c3bb8a0d679d43614b"}}}
2021-09-23 17:52:58 DEBUG (MainThread) [custom_components.myhome] Negotiating command session.
2021-09-23 17:52:58 DEBUG (MainThread) [custom_components.myhome] Reply: *#*1##
2021-09-23 17:52:58 DEBUG (MainThread) [custom_components.myhome] Received nonce: *#234123812##
2021-09-23 17:52:58 DEBUG (MainThread) [custom_components.myhome] Sending command session password.
2021-09-23 17:52:58 DEBUG (MainThread) [custom_components.myhome] Reply: *#*1##
2021-09-23 17:52:58 DEBUG (MainThread) [custom_components.myhome] Command session established.
2021-09-23 17:52:58 INFO (MainThread) [custom_components.myhome] Message *#1*38*#1*169*0## was successfully sent.
2021-09-23 17:52:58 DEBUG (MainThread) [custom_components.myhome] Received: *#1*38*1*169*4##
2021-09-23 17:52:58 INFO (MainThread) [custom_components.myhome] Light 38 is switched on at 69%.
2021-09-23 17:52:58 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: entity_id=light.table_dots, old_state=<state light.table_dots=off; supported_color_modes=['brightness'], A=3, PL=8, friendly_name=Table dots, supported_features=33 @ 2021-09-23T17:52:18.375041+03:00>, new_state=<state light.table_dots=on; supported_color_modes=['brightness'], color_mode=brightness, brightness=176, A=3, PL=8, friendly_name=Table dots, supported_features=33 @ 2021-09-23T17:52:58.934458+03:00>>
2021-09-23 17:52:58 DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection] [140345955698576] Sending {"id": 2, "type": "event", "event": {"event_type": "state_changed", "data": {"entity_id": "light.table_dots", "old_state": {"entity_id": "light.table_dots", "state": "off", "attributes": {"supported_color_modes": ["brightness"], "A": "3", "PL": "8", "friendly_name": "Table dots", "supported_features": 33}, "last_changed": "2021-09-23T14:52:18.375041+00:00", "last_updated": "2021-09-23T14:52:18.375041+00:00", "context": {"id": "b6ae20918e37399d495dcdd73b6a2da6", "parent_id": null, "user_id": null}}, "new_state": {"entity_id": "light.table_dots", "state": "on", "attributes": {"supported_color_modes": ["brightness"], "color_mode": "brightness", "brightness": 176, "A": "3", "PL": "8", "friendly_name": "Table dots", "supported_features": 33}, "last_changed": "2021-09-23T14:52:58.934458+00:00", "last_updated": "2021-09-23T14:52:58.934458+00:00", "context": {"id": "f6b34d0151a955fe04193d3a7b1cd2c3", "parent_id": null, "user_id": "627351d1043e48c3bb8a0d679d43614b"}}}, "origin": "LOCAL", "time_fired": "2021-09-23T14:52:58.934458+00:00", "context": {"id": "f6b34d0151a955fe04193d3a7b1cd2c3", "parent_id": null, "user_id": "627351d1043e48c3bb8a0d679d43614b"}}}
2021-09-23 17:52:58 DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection] [140345772026992] Sending {"id": 23, "type": "event", "event": {"event_type": "state_changed", "data": {"entity_id": "light.table_dots", "old_state": {"entity_id": "light.table_dots", "state": "off", "attributes": {"supported_color_modes": ["brightness"], "A": "3", "PL": "8", "friendly_name": "Table dots", "supported_features": 33}, "last_changed": "2021-09-23T14:52:18.375041+00:00", "last_updated": "2021-09-23T14:52:18.375041+00:00", "context": {"id": "b6ae20918e37399d495dcdd73b6a2da6", "parent_id": null, "user_id": null}}, "new_state": {"entity_id": "light.table_dots", "state": "on", "attributes": {"supported_color_modes": ["brightness"], "color_mode": "brightness", "brightness": 176, "A": "3", "PL": "8", "friendly_name": "Table dots", "supported_features": 33}, "last_changed": "2021-09-23T14:52:58.934458+00:00", "last_updated": "2021-09-23T14:52:58.934458+00:00", "context": {"id": "f6b34d0151a955fe04193d3a7b1cd2c3", "parent_id": null, "user_id": "627351d1043e48c3bb8a0d679d43614b"}}}, "origin": "LOCAL", "time_fired": "2021-09-23T14:52:58.934458+00:00", "context": {"id": "f6b34d0151a955fe04193d3a7b1cd2c3", "parent_id": null, "user_id": "627351d1043e48c3bb8a0d679d43614b"}}}
```

**Comment by anotherjulien** [comment]
OK, so your dimmer correctly responds to "100 levels" commands, but somehow it does not respond to queries for DIMENSION 4...
The issue is, when you "just" switch a light on, it only replies with a "10 levels" value (so `*1*9*38##` meaning 90%).
Since this value is approximative, I discard it and send a query for the "100 levels" value ( `*#1*38*4##` here).
The issue is that you get no reply for this DIMENSION. So you switch ON, HA expects a confirmation from the device that it's on, but with no reply, HA assumes it did not work and set the switch back to OFF.
But when you set the brightness, you get a reply (`*#1*38*1*169*4##`= 69%, nice) so HA knows what the actual status is.

What kind of gateway do you use for your system?

**Comment by akalikin1** [comment]
> What kind of gateway do you use for your system?

MH201
I will try to update to to last firmware now and will check if this issue persists.

But as I said in first message I work with this gateway in nodered, I see all commands sent to the gateway, and it works ok.

**Comment by akalikin1** [comment]
> I will try to update to to last firmware now and will check if this issue persists.

Updated, double checked everything, issue still persists.

```
Home Assistant Data

Core Version
core-2021.9.7

Version
supervisor-2021.09.0

Operating System
Home Assistant OS 6.4
```

**Comment by anotherjulien** [comment]
Sorry for the late reply, could you tell me if your light responds to the `*#1*38*1##` command?
If so, I'll change the "default" command in a future release

**Comment by akalikin1** [comment]
> responds to the `*#1*38*1##` command?

No, my dimmable light does not responds on this command
But my dimmable light respond to `*1*1*38##`  command, on this command, my light turn on with last remembered brightness

**Comment by anotherjulien** [comment]
Sorry, this was unclear; this command was not meant to turn the light on, it was meant to query the current brightness.
If the command works, then you should be able to see a message similar to `DEBUG (MainThread) [custom_components.myhome] Received: *#1*38*1*169*4##` in the logs

**Comment by akalikin1** [comment]
> you should be able to see a message

Yes, gateway responds to `*#1*38*1##`

if brightness 100% I receive `*#1*38*1*200*4##`
if brightness 20% I receive `*#1*38*1*120*4##`

**Comment by anotherjulien** [comment]
Excellent, so a temporary fix has been included in this commit 9e46f6bf77c5fe414bdff3d949bd4fae0155c2a8 pending the final fix that will need to happen in OWNd at a later date.
You can either test the current "master" version, or wait a few days for me to push the fix along with a couple other ones in a packaged release.
